packages/core/js-client/src/api.spec.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { fileURLToPath } from "url";
+
+import { compileFromPath } from "@fluencelabs/aqua-api";
+import { ServiceDef } from "@fluencelabs/interfaces";
+import { describe, expect, it } from "vitest";
+
+import { v5_registerService } from "./api.js";
+import { callAquaFunction } from "./compilerSupport/callFunction.js";
+import { withPeer } from "./util/testUtils.js";
+
+class Calc {
+  private _state: number = 0;
+
+  add(n: number) {
+    this._state += n;
+  }
+
+  subtract(n: number) {
+    this._state -= n;
+  }
+
+  multiply(n: number) {
+    this._state *= n;
+  }
+
+  divide(n: number) {
+    this._state /= n;
+  }
+
+  reset() {
+    this._state = 0;
+  }
+
+  getResult() {
+    return this._state;
+  }
+}
+
+describe("User API methods", () => {
```

Comment:
What do you think, maybe we need a simple test here that checks that inherited methods work if you use high level api? Do you have this test?

Reply:
Or you want to not allow this for high level api as well?

Reply:
Nice catch. Actually yes, i want to allow calling inherited methods in high level api. Because if not, it would be surprising for user that he passes ts validation and fails at runtime because his methods are inherited.